src/Data/Functor/Bind.hs

```diff
@@ -40,6 +42,18 @@ module Data.Functor.Bind (
 
 import Data.Functor.Apply
 import Data.Functor.Bind.Class
+import GHC.Generics
+
+-- | Generic '(>>-)'. Caveats:
+--
+--   1. Will not compile if @m@ is a sum type.
+--   2. Will not compile if @m@ contains fields that do not mention its type variable.
+--   3. Will not compile if @m@ is a recursive type.
```

Comment:
We can't easily give `(:.:)` a `Bind` instance, but we should still change this caveat slightly. How about "Will not compile if `m` contains fields where the type variable appears underneath the composition of type constructors (e.g., `f (g a)`)"?